task/memory_storage.go

```diff
@@ -9,7 +9,7 @@ import (
 func NewMemoryStorage() Storage {
 	newStorage := &memoryStorage{
 		Mutex:   sync.Mutex{},
-		Storage: map[string]*TaskObject{},
```

Comment:
this ensures we are creating copies of our objects - so modifications to the object after saving does not change it anymore. same as a stored task in redis or etcd would be